SQLite/Core/ConnectionPool.swift

```diff
@@ -0,0 +1,201 @@
+        @warn_unused_result func prepare(statement: String, _ bindings: [Binding?]) throws -> Statement { return try connection.prepare(statement, bindings) }
+        @warn_unused_result func prepare(statement: String, _ bindings: [String: Binding?]) throws -> Statement { return try connection.prepare(statement, bindings) }
+        
+        func run(statement: String, _ bindings: Binding?...) throws -> Statement { return try connection.run(statement, bindings) }
+        func run(statement: String, _ bindings: [Binding?]) throws -> Statement { return try connection.run(statement, bindings) }
+        func run(statement: String, _ bindings: [String: Binding?]) throws -> Statement { return try connection.run(statement, bindings) }
+        
+        @warn_unused_result func scalar(statement: String, _ bindings: Binding?...) -> Binding? { return connection.scalar(statement, bindings) }
+        @warn_unused_result func scalar(statement: String, _ bindings: [Binding?]) -> Binding? { return connection.scalar(statement, bindings) }
+        @warn_unused_result func scalar(statement: String, _ bindings: [String: Binding?]) -> Binding? { return connection.scalar(statement, bindings) }
+        
+        func transaction(mode: TransactionMode, block: (Connection) throws -> Void) throws { return try connection.transaction(mode, block: block) }
+        func savepoint(name: String, block: (Connection) throws -> Void) throws { return try connection.savepoint(name, block: block) }
+
+        func sync<T>(block: () throws -> T) rethrows -> T { return try connection.sync(block) }
+        func check(resultCode: Int32, statement: Statement? = nil) throws -> Int32 { return try connection.check(resultCode, statement: statement) }
+        
+    }
+    
+    
+    // Acquires a read/write connection to the database
+    
+    var writeConnectionInit = dispatch_once_t()
+    
+    public var writable : DirectConnection {
+
+        dispatch_once(&writeConnectionInit) {
+        
+            let flags = SQLITE_OPEN_CREATE | SQLITE_OPEN_READWRITE | SQLITE_OPEN_WAL | SQLITE_OPEN_NOMUTEX
+            self.writeConnection = try! DirectConnection(self.location, flags: flags, dispatcher: ReentrantDispatcher("SQLite.ConnectionPool.Write"), vfsName: vfsName)
+            self.writeConnection.busyTimeout = 2
+            
+            for setupProcessor in self.internalSetup.values {
+                try! setupProcessor(self.writeConnection)
+            }
+            
+            for setupProcessor in self.setup {
+                try! setupProcessor(self.writeConnection)
+            }
+            
+        }
+        
+        return writeConnection
+    }
+    
+    // Acquires a read only connection to the database
+    public var readable : Connection {
+        
+        var borrowed : BorrowedConnection!
+        
+        repeat {
```

Comment:
Isn't this a busy loop? If all the readers are checked out it would appear this method will just keep repeating busily until one is available. Would be a lot better to use a `dispatch_semaphore` or `NSCondition` to let it sleep and get a signal when a connection is actually available.